pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2151,16 +2166,42 @@ public boolean addNewReloadAllSegmentsJob(String tableNameWithType, String jobId
     Map<String, String> jobMetadata = new HashMap<>();
     jobMetadata.put(CommonConstants.ControllerJob.JOB_ID, jobId);
     jobMetadata.put(CommonConstants.ControllerJob.TABLE_NAME_WITH_TYPE, tableNameWithType);
-    jobMetadata.put(CommonConstants.ControllerJob.JOB_TYPE, ControllerJobType.RELOAD_ALL_SEGMENTS.toString());
+    jobMetadata.put(CommonConstants.ControllerJob.JOB_TYPE, ControllerJobType.RELOAD_SEGMENT.toString());
     jobMetadata.put(CommonConstants.ControllerJob.SUBMISSION_TIME_MS, Long.toString(System.currentTimeMillis()));
     jobMetadata.put(CommonConstants.ControllerJob.MESSAGE_COUNT, Integer.toString(numberOfMessagesSent));
-    return addControllerJobToZK(jobId, jobMetadata);
-  }
-
-  private boolean addControllerJobToZK(String jobId, Map<String, String> jobMetadata) {
-    String jobResourcePath = ZKMetadataProvider.constructPropertyStorePathForControllerJob();
+    return addControllerJobToZK(jobId, jobMetadata,
+                                ZKMetadataProvider.constructPropertyStorePathForControllerJobResource(
+                                ControllerJobType.RELOAD_SEGMENT.name()),
+                                _propertyStore);
+  }
+  public static boolean addRebalanceResultToZk(String tableNameWithType,
```

Comment:
Do we need to retry if writing to zk fails

Reply:
retries are internally taken care of in ZkBaseDataAccessor. 
https://github.com/apache/helix/blob/master/helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java#L286